mypyc/ops.py

```diff
@@ -1230,9 +1241,10 @@ class ClassIR:
 
     # TODO: Use dictionary for attributes in addition to (or instead of) list.
 
-    def __init__(self, name: str, module_name: str) -> None:
+    def __init__(self, name: str, module_name: str, callable: Optional[str] = None) -> None:
         self.name = name
         self.module_name = module_name
+        self.callable = callable
```

Comment:
Instead of having a separate field for callable, I think the cleanest thing to do might be to make the function actually be the `__call__` method of the class. Then the emitter code could figure out the name based on that, without needing a new field for it.

Reply:
Oh, and doing it that way will have the pleasant side-effect that it will make user-defined classes that provide `__call__` methods work.

Reply:
Done.